advanced.matrix/src/main/java/learning/solutions/advanced/matrix/domain/RCell.java

```diff
@@ -60,6 +64,40 @@ public double getMeanLong() {
         return meanLong;
     }
 
+    private List<Long> rewardTimestamps = new ArrayList<>();
+    private List<Long> penaltyTimestamps = new ArrayList<>();
+
+    public double recalculateRewardValue() {
+        Long currentTime = System.currentTimeMillis();
```

Comment:
You don't have to pass in the timestamp - just get System.currentTimeMillis() inside the purge calls.